### PR TITLE
Add exercises about interfaces

### PR DESCRIPTION
**Pull request by illicitonion**
Contributes to https://github.com/CodeYourFuture/immersive-go-course/issues/123

### CODE REVIEW

File: projects/interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+* If you call `b.Read` with a slice big enough to read all of the bytes in the buffer, all of the bytes are read.
+* If you call `b.Read` with a slice smaller than the contents of the buffer, some of the bytes are read. If you call it again, the next bytes are read.
+
+### Implementing our own `bytes.Buffer`
+
+Now that we have some tests for the behaviour of this struct, let's implement our own copy.
+
+Make a new struct, we'll call it `OurByteBuffer`.
+
+Change the tests we have to refer to our type instead of `bytes.Buffer`.
```
Comment (sre-is-laura) [inline projects/interfaces/README.md:58]: which should just be one line of a good table driven test, might be worth emphasising

**Review by sre-is-laura — APPROVED** [review]

**Comment by SallyMcGrath** [comment]
Boop

**Review by SallyMcGrath — APPROVED** [review]

File: projects/interfaces/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+You should write unit tests which show at least the following (you can write more if you want!):
+* If you make a buffer named `b` containing some bytes, calling `b.Buffer()` returns the same bytes you created it with.
+* If you write some extra bytes to that buffer using `b.Write()`, a call to `b.Buffer()` returns both the initial bytes and the extra bytes.
+* If you call `b.Read` with a slice big enough to read all of the bytes in the buffer, all of the bytes are read.
+* If you call `b.Read` with a slice smaller than the contents of the buffer, some of the bytes are read. If you call it again, the next bytes are read.
+
+### Implementing our own `bytes.Buffer`
+
+Now that we have some tests for the behaviour of this struct, let's implement our own copy.
+
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply projects/interfaces/README.md:55]: Added!

**Comment by SallyMcGrath** [comment]
@illicitonion did you forget this one btw?

**Comment by illicitonion** [comment]
I have a few small things to finish off on it; discussed them with Laura this week and will pick it up soon!

File: projects/interfaces/README.md (hunk 1 above)
Comment (illicitonion) [reply projects/interfaces/README.md:58]: I added a long chunk of exposition about this to the sample implementation - PTAL there in https://github.com/CodeYourFuture/immersive-go-course/pull/131/commits/cc9349f41d587cdc88a342c1f6ec3ac138f55d8a

**Pull request merged**